src/mpcCoreKit.ts

```diff
@@ -292,30 +293,36 @@ export class Web3AuthMPCCoreKit implements ICoreKit, IMPCContext {
       tssShares.push(tssShare);
     }
 
-    const tssCurve = getKeyCurve(this.keyType);
+    const tssCurve = getKeyCurve(keyType);
     const finalKey = lagrangeInterpolation(tssCurve, tssShares, tssIndexesBN);
     // reset instance after recovery completed
     await this.init();
     return finalKey.toString("hex", 64);
   }
 
+  public getServiceProviderKeyType() {
```

Comment:
This does not access the service provider. Why is it called `getServiceProviderKeyType`?
Add function comment to explain further.

Reply:
Why is this a public function?

Reply:
It is used to decide what service provider's keyType that mpc corekit should instantiate during init.

in multicurve, it is defaulted to secp256k1 only
in legacy, it is depend on the keyType that is configured 

will move to private